pyrobosim/pyrobosim/navigation/rrt.py

```diff
@@ -164,28 +172,31 @@ def plan(self, start, goal):
                 return
 
         # Now back out the path
-        if self.bidirectional:
+        if self.bidirectional and not direct_connect:
             n = n_goal_start_tree
         else:
             n = n_goal
-        path_poses = [n.pose]
-        path_built = False
-        while not path_built:
-            if n.parent is None:
-                path_built = True
-            else:
-                n = n.parent
-                path_poses.append(n.pose)
-        path_poses.reverse()
-        if self.bidirectional:
-            n = n_goal_goal_tree
+        if not direct_connect:
+            path_poses = [n.pose]
             path_built = False
             while not path_built:
                 if n.parent is None:
                     path_built = True
                 else:
                     n = n.parent
                     path_poses.append(n.pose)
+            path_poses.reverse()
+            if self.bidirectional:
+                n = n_goal_goal_tree
+                path_built = False
+                while not path_built:
+                    if n.parent is None:
+                        path_built = True
+                    else:
+                        n = n.parent
+                        path_poses.append(n.pose)
+        else:
+            path_poses = [n_start.pose, direct_goal.pose]
```

Comment:
I'm not a fan of all this added code for something that should be simple.

Maybe consider something like this: In the first block of code earlier in the file that checks for a direct path from start to goal, can we just immediately return the path up there, and do away with all these additions?

Reply:
Yes, that is definitely doable.
I was going for a minimal return statement philosophy. 
But it does look like a lot of changes for a simple addition.
Thanks !